datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -470,6 +499,24 @@ protected int constructExpectedNumberOfTasks(DatastreamGroup dg, List<String> in
     return expectedNumberOfTasks;
   }
 
+  private void updateOrRegisterElasticTaskAssignmentMetrics(DatastreamGroupPartitionsMetadata datastreamPartitions,
+      int totalTaskCount) {
+    int totalPartitions = datastreamPartitions.getPartitions().size();
+    int actualPartitionsPerTask = (totalPartitions / totalTaskCount)
+        + (((totalPartitions % totalTaskCount) == 0) ? 0 : 1);
+
+    int partitionsPerTask = resolveConfigWithMetadata(datastreamPartitions.getDatastreamGroup(),
+        CFG_PARTITIONS_PER_TASK, _partitionsPerTask);
+    String taskPrefix = datastreamPartitions.getDatastreamGroup().getTaskPrefix();
+    ElasticTaskAssignmentInfo elasticTaskAssignmentInfo =
+        new ElasticTaskAssignmentInfo(actualPartitionsPerTask, actualPartitionsPerTask > partitionsPerTask);
+
+    if (!_elasticTaskAssignmentInfoHashMap.containsKey(taskPrefix)) {
+      registerElasticTaskAssignmentMetrics(taskPrefix);
+    }
+    _elasticTaskAssignmentInfoHashMap.put(taskPrefix, elasticTaskAssignmentInfo);
```

Comment:
Do we not need to remove the taskPrefix from this Map when not in use.

Reply:
I need to expose this map in StickyMulticastStrategy (where I can find out that a datastream group is missing). Partition assignment is called per group, so there is no way to know that a datastream group is deleted. Is that okay with you?

Reply:
done